### DISCUSSION TITLE
Welcome to orbit Discussions!

### DISCUSSION THREAD

**Discussion by bijington** [bijington/orbit#6] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by ijump** [comment]
Hi, 

I look forward to trying out the Orbit game engine. I like the cleanness and simplicity. And I like how it does not try to reinvent anything that Maui already provides. 

I was looking for an engine/library that could mix enterprise application development with game development. Maui/Orbit looks to be a great fit. My initial prototypes are going to be heavily text/menu based. But later, I want the ability to expand into more graphical/gaming UI and interaction. I'm not sure exactly what it will look like. Working on a first demo.

Not sure how, but maybe I can contribute along the way.

Thank you
Steve

**Reply by bijington** [reply]
Hi Steve,

Thank you for the compliment I really appreciate it!

While I am hoping to not provide things that MAUI already offers I have been playing around with being able to create a Text based GameObject and add it to a scene, the thought being that one could define some behaviour such as animating the opacity or location without too much effort. I am hopeful we could simply build this behaviour fluently, add it to the scene and then let it manage the complexity of updating the animation based on `Update` and `Render` calls. Does this sound like it could be of use to you?

I really hope the prototyping goes well, please feel free to reach out for input or provide details of you are using it to help improve the engine 😄

**Reply by ijump** [reply]
Hi Shaun,

That sounds very useful! I was thinking about advanced text rendering. 

Quick intro to my project. It's basically going to be a more advanced choose your adventure platform. There will be two main apps - a writer and a reader. The reader can also be combined and shipped with specific content as an individual app/game/story itself. I think I've made all the initial decisions with the different layers of the stack. And I also finished building the db schema. I'm using https://www.smarterasp.net/ to host the sql database. Very happy with them so far for cheap prototyping hosting. I made an initial prototype of "writer" in blazor, but am moving it over to Maui. 

Then on to the reader. Initially, this will be basic display of text, but then I want to start getting fancy with it. That's where your engine will come into play. Not exactly sure how yet. I just know I want to break out of the Maui component mold.

So, yes! Very interested in a Text based GameObject. Let me know if I can help. 

Thanks
Steve

**Reply by bijington** [reply]
Apologies for the delayed response.

It sounds like you are progressing well and I would love to see the outcome of this!

My current plan is to build some slides for a talk through this engine which might be a bit of a gimmick but I'll enjoy it. For that I certainly need text and it felt right to make this task as easy as possible. 

I would be more than keen for some help and that can be in any form. I have something although it feels a little rough around the edges, perhaps I can put it up on a branch and you see if it feels right or if there is something better we could achieve?

**Reply by ijump** [reply]
That sounds great! 

I got pulled away doing taxes and other personal admin stuff.

I also have to teach a workshop coming up in two weeks on open source desktop publishing. I'm going to put some emphasis on using the asciidoc format. I think my application will eventually output asciidoc/pdf for print versions of the storygames. I guess that brings up for me a question of how to store text layout/animation data that would translate to both your game engine and to print versions. Just something to daydream about for the moment. The main emphasis would be the digital interaction. Print would just be a bonus feature.

If you create the branch and point me to a place you would like me to look at, that would be great. I look forward to helping (if I can). Over the years, I've had a varied skillset, so hopefully I can be helpful.
